### PR TITLE
Add functional test

### PR DESCRIPTION
**Pull request by pleshakov**
### Proposed changes

Problem:
Ensure the exporter library correctly sends telemetry data via OTel

Solution:
Add a functional test:
- Exporter sends telemetry data to an OTel collector
- Ensure the collector successfully receives the data by looking at its logs
- Include the test into GitHub pipeline.

CLOSES https://github.com/nginxinc/telemetry-exporter/issues/7

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/telemetry-exporter/blob/main/CONTRIBUTING.md)
      guide
- [x] I have proven my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have ensured the README is up to date
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch on my own fork

### CODE REVIEW

**Review by ciarams87 — APPROVED** [review]

File: tests/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# syntax=docker/dockerfile:1.6
+# this is here so we can grab the latest version of the collector and have dependabot keep it up to date
```
[the remark this answers is not in the document]
Comment (pleshakov) [reply tests/Dockerfile:2]: @lucacome we read that Dockerfile here -- https://github.com/nginxinc/telemetry-exporter/blob/8274462eaf00342cf9a2cd8ec171aeb30a8ae373/tests/exporter_test.go#L70

**Comment by pleshakov** [comment]
@ciarams87 could you possibly double check these commits https://github.com/nginxinc/telemetry-exporter/pull/26/files/8274462eaf00342cf9a2cd8ec171aeb30a8ae373..fb1fba20f17e32957b3df33ef4a2f1e7407e2e9e  - those were the results of merging Luca suggestions

**Review by sjberman — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -64,5 +64,20 @@ jobs:
         uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
         with:
           name: cover-${{ github.run_id }}.html
-          path: ${{ github.workspace }}/cmd-cover.html
+          path: ${{ github.workspace }}/cover.html
```
Comment (ciarams87) [inline .github/workflows/ci.yml:67]: If we want, we can upload both coverage reports. I think this should work:
```suggestion
          path: |
            ${{ github.workspace }}/cover.html
            ${{ github.workspace }}/generator-cover.html
```

**Comment by ciarams87** [comment]
> @ciarams87 could you possibly double check these commits https://github.com/nginxinc/telemetry-exporter/pull/26/files/8274462eaf00342cf9a2cd8ec171aeb30a8ae373..fb1fba20f17e32957b3df33ef4a2f1e7407e2e9e - those were the results of merging Luca suggestions

That looks fine. One non-blocking suggestion around uploading both unit-test coverage reports, otherwise looks good!

**Pull request merged**